code/go/internal/validator/semantic/validate_docs_structure.go

```diff
@@ -0,0 +1,222 @@
+	"io/fs"
+	"path"
+	"strings"
+
+	spec "github.com/elastic/package-spec/v3"
+	"github.com/elastic/package-spec/v3/code/go/internal/fspath"
+	"github.com/elastic/package-spec/v3/code/go/internal/pkgpath"
+	"github.com/elastic/package-spec/v3/code/go/pkg/specerrors"
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/text"
+	"gopkg.in/yaml.v3"
+)
+
+// EnforcedSections represents the enforced documentation structure configuration.
+type EnforcedSections struct {
+	Sections []Section `yaml:"enforced_sections"`
+}
+
+// Section represents a section in the enforced documentation structure.
+type Section struct {
+	Name        string `yaml:"name"`
+	Description string `yaml:"description"`
+}
+
+// ValidateDocsStructure validates the structure of documentation files against enforced sections.
+func ValidateDocsStructure(fsys fspath.FS) specerrors.ValidationErrors {
+	config, err := shouldValidateDocsStructure(fsys)
+	if err != nil {
+		return specerrors.ValidationErrors{
+			specerrors.NewStructuredError(
+				fmt.Errorf("failed to determine if documentation structure validation should be performed: %w", err),
+				specerrors.UnassignedCode,
+			),
+		}
+	}
+	if config == nil {
+		return nil
+	}
+
+	enforcedSections, err := readDocsStructureEnforcedConfig(fsys, config)
+	if err != nil {
+		return specerrors.ValidationErrors{
+			specerrors.NewStructuredError(
+				fmt.Errorf("failed to read enforced documentation structure configuration: %w", err),
+				specerrors.UnassignedCode,
+			),
+		}
+	}
+
+	err = validateReadmeStructure(fsys, enforcedSections)
+
+	if err != nil {
+		return specerrors.ValidationErrors{
+			specerrors.NewStructuredError(
+				err,
+				specerrors.UnassignedCode,
+			),
+		}
+	}
+	return nil
+}
+
+func shouldValidateDocsStructure(fsys fspath.FS) (*specerrors.DocsStructureEnforced, error) {
+	validationPath := "validation.yml"
+	files, err := pkgpath.Files(fsys, validationPath)
+	if err != nil || len(files) == 0 {
+		return nil, nil
+	}
+
+	validationFile := files[0]
+	data, err := validationFile.ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file %s: %w", validationFile.Path(), err)
+	}
+
+	var config specerrors.ConfigFilter
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal YAML from %s: %w", validationFile.Path(), err)
+	}
+
+	if config.DocsStructureEnforced.Enabled {
+		return &config.DocsStructureEnforced, nil
+	}
+
+	return nil, nil
+}
+
+func readDocsStructureEnforcedConfig(fsys fspath.FS, config *specerrors.DocsStructureEnforced) ([]string, error) {
+	defaultSections, err := loadSectionsFromConfig(fmt.Sprintf("%d", config.Version))
+	if err != nil {
+		return nil, fmt.Errorf("failed to load enforced sections from config: %w", err)
+	}
+
+	var sections []string
+	for _, section := range defaultSections {
+		if contains(config.DocsStructureSkip, section) {
+			continue
+		}
+		sections = append(sections, section)
```

Comment:
I think It could be used `slices.ContainsFunc()` here:
```go
	for _, section := range defaultSections {
		if !slices.ContainsFunc(config.DocsStructureSkip, func(s specerrors.DocsStructureSkip) bool {
			return s.Title == section
		}) {
			sections = append(sections, section)
		}
	}
```
or keeping the same condition:
```go
	for _, section := range defaultSections {
		if slices.ContainsFunc(config.DocsStructureSkip, func(s specerrors.DocsStructureSkip) bool {
			return s.Title == section
		}) {
			continue
		}
		sections = append(sections, section)
	}
```

WDYT?

If that makes sense, the `contains` function from line 115 could be removed